nipype/interfaces/dtitk/utils.py

```diff
@@ -1,367 +1,302 @@
+Elman, L.B., Melhem, E.R., Gee, J.C., High-dimensional spatial normalization of
+diffusion tensor images improves the detection of white matter differences in
+amyotrophic lateral sclerosis, IEEE Transactions on Medical Imaging,
+26(11):1585-1597, November 2007. PMID: 18041273.
 
-class TVAdjustOriginTask(CommandLineDtitk):
-    """
-    Moves the origin of a tensor volume to zero
-
-    Example
-    -------
+The original piecewise-affine tensor-based DTI registration algorithm at the
+core of DTI-TK
 
-    >>> import nipype.interfaces.dtitk as dtitk
-    >>> node = dtitk.TVAdjustOriginTask()
-    >>> node.inputs.in_file = 'diffusion.nii'
-    >>> node.run() # doctest: +SKIP
-    """
+Zhang, H., Yushkevich, P.A., Alexander, D.C., Gee, J.C., Deformable
+registration of diffusion tensor MR images with explicit orientation
+optimization, Medical Image Analysis, 10(5):764-785, October 2006. PMID:
+16899392.
 
-    input_spec = TVAdjustOriginInputSpec
-    output_spec = TVAdjustOriginOutputSpec
-    _cmd = 'TVAdjustVoxelspace'
-    _suffix = "_originzero"
+"""
+__author__ = 'kjordan'
 
-    def _list_outputs(self):
-        outputs = self.output_spec().get()
-        outputs['out_file'] = self.inputs.out_file
-        if not isdefined(self.inputs.out_file):
-            outputs["out_file"] = self._gen_fname(self.inputs.in_file,
-                                                  suffix=self._suffix,
-                                                  ext='.'+'.'.join(
-                                                      self.inputs.in_file.
-                                                      split(".")[1:]))
-        outputs["out_file"] = os.path.abspath(outputs["out_file"])
-        return outputs
+from ..base import TraitedSpec, CommandLineInputSpec, File, traits, isdefined
+from ...utils.filemanip import fname_presuffix
+from .base import CommandLineDtitk
+import os
 
-    def _gen_filename(self, name):
-        if name == "out_file":
-            return self._list_outputs()["out_file"]
-        return None
+__docformat__ = 'restructuredtext'
 
 
 class TVAdjustVoxSpInputSpec(CommandLineInputSpec):
-    in_file = File(desc="image to resample", exists=True, mandatory=True,
-                   position=0, argstr="-in %s")
-    out_file = traits.Str(genfile=True, desc='output path', position=1,
-                          argstr="-out %s")
-    origin = traits.Str(desc='xyz voxel size', exists=True, mandatory=False,
-                        position=4, argstr='-origin %s')
-    target = traits.Str(desc='target volume', exists=True, mandatory=False,
-                        position=2, argstr="-target %s")
-    vsize = traits.Str(desc='resampled voxel size', exists=True,
-                       mandatory=False, position=3, argstr="-vsize %s")
+    in_file = File(desc="tensor volume to modify", exists=True,
+                   mandatory=True, argstr="-in %s")
+    out_file = File(genfile=True, desc='output path',
```

Comment:
I think `genfile` is an older mechanism than `name_template`, and no longer recommended. So we'd just drop `genfile=True` and keep this trait otherwise as is. @satra is this correct?

Reply:
We still want to use `genfile=True` for `TVtool`, right? (this is one that we are building the filename from multiple inputs)

Reply:
Yes, that one is fine to leave, since we're not modifying it to use `name_template`.